### PR TITLE
Fix namespace font locking

### PR DESCRIPTION
**Pull request by Bost**
- [x] The commits are consistent with our [contribution guidelines][1]
- [x] You've added tests (if possible) to cover your change(s). Indentation & font-lock tests are extremely important!
- [ ] All tests are passing (`make test`)

- Using cask for testing is a pain :-(
1. Cask installation doesn't work from behind a proxy
2. Being behind or a proxy or not - in both cases I get:
```
$ make test
make: *** No rule to make target '/home/bost/dev/clojure-mode/.cask/25.3/elpa', needed by 'test'.  Stop.
```
However `M-x spacemacs/ert-run-tests-buffer` shows no new errors (on top of those 8 existing already)

- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)
- I fixed a bug. No readme change is needed.

### CODE REVIEW

File: test.clj [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+  (:require
+   [clojure.something]
+   [something.s]
+   [oneword]
+   ;; TODO mixedCase shouldn't be font-locked to clojure-interop-method-face
+   [mixedCase]
+   [CamelCase]))
+
+;; examples of valid namespace definitions
+(comment
```
Comment (bbatsov) [inline test.clj]: I'd actually also add new unit tests verifying the validity of your changes.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 * [#443](https://github.com/clojure-emacs/clojure-mode/issues/443): Fix behavior of `clojure-forward-logical-sexp` and `clojure-backward-logical-sexp` with conditional macros.
 * [#429](https://github.com/clojure-emacs/clojure-mode/issues/429): Fix a bug causing last occurrence of expression sometimes is not replaced when using `move-to-let`.
 * [#423](https://github.com/clojure-emacs/clojure-mode/issues/423): Make `clojure-match-next-def` more robust against zero-arity def-like forms.
+* Fix namespace font-locking
```
Comment (bbatsov) [inline CHANGELOG.md]: You can also mention here the GH issue you've fixed. This sentence should end with a `.`.

Comment (Bost) [reply CHANGELOG.md]: I did the fix out of pure excitement of having found "a pristine spot on a wall I can spray my tag on". Ok seriously I there is no GH issue for my PR. Although the #280 seems to be closely related.

**Comment by bbatsov** [comment]
> However M-x spacemacs/ert-run-tests-buffer shows no new errors (on top of those 8 existing already)

Hmm, that's odd. There shouldn't be any failing tests (there were certainly none last time I checked). Unfortunately the TravisCI setup seems to have been magically broken and I don't have time to look into this myself at this time.

**Comment by Bost** [comment]
> Hmm, that's odd. There shouldn't be any failing tests (there were certainly none last time I checked). 

It seems like something's messed up somewhere also on my side. `M-x spacemacs/ert-run-tests-buffer` on current master 35f5d71 now gives me only 4 errors. Not 8 as I reported. And my changes introduced 4 new errors on top of the 4 from the master. Merde :-( Could you please run the test on your machine? Thx.

Here the `M-x spacemacs/ert-run-tests-buffer` of `test/clojure-mode-font-lock-test.el` on current master 35f5d71:
```
Selector: (satisfies (lambda (test) (eq cbuf (spacemacs//find-ert-test-buffer test))))
Passed:  30
Failed:  4 (4 unexpected)
Skipped: 0
Total:   34/34

Started at:   2017-10-13 01:07:49+0200
Finished.
Finished at:  2017-10-13 01:07:52+0200

F.F.......F.............F.........

F clojure-mode-syntax-table/characters
    (ert-test-failed
     ((should
       (eq
        (clojure-test-face-at 1 2 "\\1")
        'clojure-character-face))
      :form
      (eq various-faces clojure-character-face)
      :value nil))

F clojure-mode-syntax-table/comment-macros
    (ert-test-failed
     ((should
       (equal
        (clojure-test-face-at 5 41 "#_ 
;; some crap
 (lala 0101
 lao

 0 0i)")
        'font-lock-comment-face))
      :form
      (equal various-faces font-lock-comment-face)
      :value nil :explanation
      (different-atoms various-faces font-lock-comment-face)))

F clojure-mode-syntax-table/fn
    (ert-test-failed
     ((should
       (eq
        (clojure-test-face-at 2 3)
        'font-lock-keyword-face))
      :form
      (eq t font-lock-keyword-face)
      :value nil))

F clojure-mode-syntax-table/ns-macro
    (ert-test-failed
     ((should
       (eq
        (clojure-test-face-at 1 10 "[ns name]")
        nil))
      :form
      (eq various-faces nil)
      :value nil))
```

**Comment by Bost** [comment]
I'm closing this PR. I'm gonna send you a new PR without regressions.

**Pull request closed without merging**